### ISSUE TITLE
Error parsing turtle from DBpedia

### ISSUE THREAD

**Issue by retog** [RubenVerborgh/N3.js#129]
Trying to parse the document at http://dbpedia.org/data/Bern.ttl 

```
N3Lexer.js:364 Uncaught (in promise) Error: 
Unexpected "dbr:Germany_at_the_2006–08_European_Nations_Cup" on line 86.
    at s._syntaxError (N3Lexer.js:364)
    at g (N3Lexer.js:331)
    at s._tokenizeToEnd (N3Lexer.js:316)
    at a.<anonymous> (N3Lexer.js:405)
    at a.r.emit (events.js:81)
    at w (_stream_readable.js:291)
    at _ (_stream_readable.js:276)
    at a.m.push (_stream_readable.js:245)
    at index.js:18
    at p.run (browser.js:153)
    at d (browser.js:123)
```

I don't see why the CURIE `dbr:Germany_at_the_2006–08_European_Nations_Cup` is unexpected, the context (lines 85-87) is as follows:
```
dbr:History_of_the_Netherlands_national_football_team	dbp:stadium	dbr:Bern .
dbr:Germany_at_the_2006–08_European_Nations_Cup	dbp:stadium	dbr:Bern .
dbr:Herman_Bemberg	dbo:deathPlace	dbr:Bern .
```

**Comment by RubenVerborgh** [comment]
Interestingly, both N3.js and serd have a problem earlier in the `Bern.ttl` file:

```
Error: Unexpected "dbr:Swiss_Medical_Students’_Association" on line 998.
```

The problem in this case is [right single quote (`\u2019`)](https://www.fileformat.info/info/unicode/char/2019/index.htm), which is not in [`PN_CHARS_BASE`](https://www.w3.org/TR/turtle/#grammar-production-PN_CHARS_BASE).

The problem in your cases is the [en dash (`\u2013`)](https://www.fileformat.info/info/unicode/char/2013/index.htm) in between `2006–08`, which, unlike the hyphen, is also not allowed in `PN_CHARS_BASE`. Note that `PN_CHARS_BASE` has a break between `\u200e` and `\u2069`, which explicitly excludes these characters.

In other words, N3.js and serd are right to reject this input, and the bug is in the Virtuoso Turtle serializer.

**Comment by retog** [comment]
That was fast, thanks!
(This confirms that its better to access DBpedia via sparql as in the default configuration of http://rdf2h-browser.linked.solutions/)

**Comment by RubenVerborgh** [comment]
No, it confirms that you should [get N-Triples instead](https://github.com/LinkedDataFragments/Server.js/blob/v2.2.2/lib/datasources/SparqlDatasource.js#L45-L47) :wink:

**Comment by retog** [comment]
Right, could implement that in the fetcher: if parsing fails, remove that format from accept header and try again... It will end with RDFa were it will most likely fail too ;)

**Comment by RubenVerborgh** [comment]
It's a genuine problem actually: the Bern.ttl file is generated at runtime as the result of a SPARQL query, so you'll keep on running into this. Haven't had any syntax problems with N-Triples so far.

**Comment by retog** [comment]
It might be good to let @kidehen know. But in between, is there any reason not to be more liberal in what N3 accepts?

**Comment by kidehen** [comment]
@retog @RubenVerborgh , 

Noted, will pass https://github.com/RubenVerborgh/N3.js/issues/129#issuecomment-374506542 along to my team.

**Comment by RubenVerborgh** [comment]
> is there any reason not to be more liberal in what N3 accepts?

The [RDF 1.1 Test Cases](https://www.w3.org/TR/rdf11-testcases/) require the parser to reject invalid Turtle. A more liberal parser is [Graphy](https://github.com/blake-regalia/graphy.js#performance).

**Comment by retog** [comment]
The n-triples work-around doesn't seem to work as I get the HTML when asking for n-triples

```
curl -D - "http://dbpedia.org/resource/Bern" -H "Accept: application/n-triples" 
HTTP/1.1 303 See Other
Date: Tue, 20 Mar 2018 19:33:42 GMT
Content-Type: text/html; charset=UTF-8
Content-Length: 0
Connection: keep-alive
Server: Virtuoso/07.20.3226 (Linux) i686-generic-linux-glibc212-64  VDB
Location: http://dbpedia.org/page/Bern
```
/CC @kidehen (see message on gitter OT)

**Comment by RubenVerborgh** [comment]
Ah, that's too bad.

**Comment by kidehen** [comment]
This is a content-negotiation related regression that will be resolved quickly.

**Comment by TallTed** [comment]
@retog /cc @RubenVerborgh 

I believe you'll find that the original issue with http://dbpedia.org/data/Bern.ttl has now been resolved (the two problematic URIs are now `<http://dbpedia.org/resource/Germany_at_the_2006\u201308_European_Nations_Cup>` and `<http://dbpedia.org/resource/Swiss_Medical_Students\u2019_Association>`), and I think this should satisfy for the n-triples workaround, if not --

```bash
$ curl -LI  "http://dbpedia.org/resource/Bern" -H "Accept: application/n-triples" 
HTTP/1.1 303 See Other
Date: Tue, 17 Apr 2018 19:37:26 GMT
Content-Type: application/n-triples
Content-Length: 0
Connection: keep-alive
Server: Virtuoso/07.20.3228 (Linux) i686-generic-linux-glibc25-64  VDB
TCN: choice
Vary: negotiate,accept
Alternates: {"/data/Bern.atom" 0.500000 {type application/atom+xml}}, {"/data/Bern.jrdf" 0.600000 {type application/rdf+json}}, {"/data/Bern.jsod" 0.500000 {type application/odata+json}}, {"/data/Bern.json" 0.600000 {type application/json}}, {"/data/Bern.jsonld" 0.500000 {type application/ld+json}}, {"/data/Bern.n3" 0.800000 {type text/n3}}, {"/data/Bern.nt" 0.800000 {type text/rdf+n3}}, {"/data/Bern.ntriples" 0.500000 {type application/n-triples}}, {"/data/Bern.ttl" 0.700000 {type text/turtle}}, {"/data/Bern.xml" 0.950000 {type application/rdf+xml}}
Link: <http://creativecommons.org/licenses/by-sa/3.0/>;rel="license",<http://mementoarchive.lanl.gov/dbpedia/timegate/http://dbpedia.org/resource/Bern>; rel="timegate"
Location: http://dbpedia.org/data/Bern.ntriples
Expires: Tue, 24 Apr 2018 19:37:26 GMT
Cache-Control: max-age=604800
Access-Control-Allow-Origin: *
Access-Control-Allow-Credentials: true
Access-Control-Allow-Methods: HEAD, GET, POST, OPTIONS
Access-Control-Allow-Headers: DNT,X-CustomHeader,Keep-Alive,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type,Accept-Encoding

HTTP/1.1 200 OK
Date: Tue, 17 Apr 2018 19:37:26 GMT
Content-Type: application/n-triples; charset=UTF-8
Content-Length: 76320
Connection: keep-alive
Server: Virtuoso/07.20.3228 (Linux) i686-generic-linux-glibc25-64  VDB
Expires: Tue, 24 Apr 2018 19:37:26 GMT
Link: <http://creativecommons.org/licenses/by-sa/3.0/>;rel="license",<http://dbpedia.org/data/Bern.xml>; rel="alternate"; type="application/rdf+xml"; title="Structured Descriptor Document (RDF/XML format)", <http://dbpedia.org/data/Bern.n3>; rel="alternate"; type="text/n3"; title="Structured Descriptor Document (N3/Turtle format)", <http://dbpedia.org/data/Bern.json>; rel="alternate"; type="application/json"; title="Structured Descriptor Document (RDF/JSON format)", <http://dbpedia.org/data/Bern.ntriples>; rel="alternate"; type="application/atom+xml"; title="OData (Atom+Feed format)", <http://dbpedia.org/data/Bern.ntriples>; rel="alternate"; type="application/odata+json"; title="OData (JSON format)", <http://dbpedia.org/page/Bern.ntriples>; rel="alternate"; type="text/html"; title="XHTML+RDFa", <http://dbpedia.org/resource/Bern.ntriples>; rel="http://xmlns.com/foaf/0.1/primaryTopic", <http://dbpedia.org/resource/Bern.ntriples>; rev="describedby", <http://mementoarchive.lanl.gov/dbpedia/timegate/http://dbpedia.org/data/Bern.ntriples>; rel="timegate"
X-SPARQL-default-graph: http://dbpedia.org
Cache-Control: max-age=604800
Access-Control-Allow-Origin: *
Access-Control-Allow-Credentials: true
Access-Control-Allow-Methods: HEAD, GET, POST, OPTIONS
Access-Control-Allow-Headers: DNT,X-CustomHeader,Keep-Alive,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type,Accept-Encoding
Accept-Ranges: bytes
```

If you hit anything else like that with Virtuoso, please do [open issues on the project](https://github.com/openlink/virtuoso-opensource/issues/)!
